src/Keboola/S3Extractor/Finder.php

```diff
@@ -171,7 +171,15 @@ private function listFilesInPrefix(): array
                     $dst = $this->subFolder . basename($object['Key']);
                 }
 
-                $filesToDownload[] = new File($this->bucket, $object['Key'], $object['LastModified'], (int)$object['Size'], $dst);
+                $file = new File($this->bucket, $object['Key'], $object['LastModified'], (int)$object['Size'], $dst);
+                $filesToDownloadCount++;
+
+                if ($this->isFileOld($file)) {
+                    continue;
+                }
+                $newFilesCount++;
```

Comment:
Predtym sa novy subory filtrovali tak, ze sa znova prechadzalo cele array.

Teraz sa to filtruje uz tu, pri iteracii vysledkov z API.

Inak, toto samo o sebe usetri kopec `memory`, pri incremental fetching (aj bez ostatných zmien).
Keďže dalej idú už iba nové súbory.

Pri full load to ale samozrejme nepomôže.